File: python/example/sort/tree/pure/node.py
import otterop.lang.array
import example.sort.tree.node

class Node:
    def __init__(self, value, children):
        _children_array = [None] * len(children)
        for i in range(len(children)):
            _children_array[i] = children[i].unwrap()
        _children = otterop.lang.array.Array.wrap(_children_array)
        self.otterop = example.sort.tree.node.Node(value, _children)

    def value(self):
        ret_otterop = self.otterop.value()
        ret = ret_otterop
        return ret


    def child(self, i):
        ret_otterop = self.otterop.child(i)
        if ret_otterop is self.otterop:
            return self
        ret = example.sort.tree.pure.node.Node.wrap(ret_otterop)
        return ret


    def children(self):
        ret_otterop = self.otterop.children()
        ret = [None] * ret_otterop.size()
        for i in range(len(ret)):
            ret_i = ret_otterop.get(i)
            if ret_i is self.otterop:
                ret[i] = self
                continue
            ret[i] = example.sort.tree.pure.node.Node.wrap(ret_i)
        return ret

    def unwrap(self):
        return self.otterop

    @staticmethod
    def wrap(wrapped):
        ret = example.sort.tree.pure.node.Node.__new__(example.sort.tree.pure.node.Node)
        ret.otterop = wrapped
        return ret

File: go/example/sort/tree/node/pure/node.go
package pure

import (
    otteropnode "github.com/otterop/example-sort/go/example/sort/tree/node"
    otteroparray "github.com/otterop/otterop/go/lang/array"
)

type Node struct {
    otterop *otteropnode.Node
}


func Wrap(otterop *otteropnode.Node) *Node {
    ret := new(Node)
    ret.otterop = otterop
    return ret
}


func NewNode(value int, children []*Node) *Node {
    var _childrenArray = make([]*otteropnode.Node, len(children))
    for i := 0; i < len(children); i++ {
        _childrenArray[i] = children[i].Unwrap()
    }
    var _children = otteroparray.Wrap(_childrenArray);
    ret := new(Node)
    ret.otterop = otteropnode.NewNode(value, _children)
    return ret
}



func (this *Node) Value() int {
    retOtterop := this.otterop.Value()
    ret := retOtterop
    return ret
}



func (this *Node) Child(i int) *Node {
    retOtterop := this.otterop.Child(i)
    if retOtterop == this.otterop { return this }
    ret := Wrap(retOtterop);
    return ret
}



func (this *Node) Children() []*Node {
    retOtterop := this.otterop.Children()
    ret := make([]*Node, retOtterop.Size())
    for i := 0; i < retOtterop.Size(); i++ {
        retI := retOtterop.Get(i)
        if retI == this.otterop {
            ret[i] = this
            continue
        }
        ret[i] = Wrap(retI)
    }
    return ret
}

func (this *Node) Unwrap() *otteropnode.Node {
    return this.otterop
}
